src/surveys/components/RangeResponse.tsx

```diff
@@ -0,0 +1,130 @@
+import classnames from 'classnames';
+import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+import { Button, Header } from '../../components';
+import {
+  ID,
+  Question,
+  QuestionOption,
+  RangeSettings,
+  SurveyAnswer,
+  Theme,
+} from '../../types';
+import { hexToRgba } from '../../utils';
+import RatingIcon from './RatingIcon';
+
+interface RangeResponseProps {
+  question: Question;
+  onAnswered: (answers: SurveyAnswer[]) => void;
+  theme?: Theme;
+}
+
+function RangeResponse({ question, onAnswered, theme }: RangeResponseProps) {
+  const [value, setValue] = useState(0);
+  const [answerId, setAnswerId] = useState<ID | null>(null);
+
+  const lightAnswerColor = theme?.answer && hexToRgba(theme.answer, 0.1);
+
+  useEffect(() => {
+    const answerQuestion = () => {
+      if (value === 0) return;
+
+      const answer = answerId
+        ? { type: question.type, answerId: answerId }
+        : { type: question.type, answer: value.toString() };
+
+      onAnswered([answer]);
+    };
+
+    const timeoutId = setTimeout(answerQuestion, 500);
```

Comment:
@AfeezGL Why is there a timeout here? The UI will feel laggy and I don't it's necessary.